### PR TITLE
1.12.3 Changelog

### PR DESCRIPTION
**Pull request by vieux**
ping @thaJeztah

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,69 @@ information on the list of deprecated flags and APIs please have a look at
...
+After making those changes, run `sudo systemctl daemon-reload`, and `sudo
+systemctl restart docker` to reload changes and (re)start the docker daemon.
+
+
+### Runtime
+
+- Prevent a deadlock in libcontainerd for Windows [#27136](https://github.com/docker/docker/pull/27136)
+- Fix error reporting in CopyFileWithTar [#27075](https://github.com/docker/docker/pull/27075)
+* Reset health status to starting when a container is restarted [#27387](https://github.com/docker/docker/pull/27387)
+* Remount everything as private in new mntns [#27609](https://github.com/docker/docker/pull/27609)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Should probably describe what it's fixing as well; Also should this be in the "Builder" section?

```
* Fix an error when trying to `ADD` or `COPY` files to the root  (`/`) of the container's filesystem
```

@tonistiigi SGTY? ^^

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,69 @@ information on the list of deprecated flags and APIs please have a look at
...
+- Fix docker exec, Fix user permissions in containers [#27610](https://github.com/docker/docker/pull/27610)
+
+### Swarm Mode
+
+- Fix conversion of restart-policy [#27062](https://github.com/docker/docker/pull/27062)
+* Vendor Swarmkit [#27554](https://github.com/docker/docker/pull/27554)
+ * Avoid restarting a task that has already been restarted [docker/swarmkit#1305]
```
Comment (thaJeztah) [inline CHANGELOG.md]: Missing https://github.com/ here and couple below

**Review by thaJeztah — COMMENTED** [review]
couple of nits / suggestions, but LGTM otherwise

File: CHANGELOG.md (hunk 1 above)
Comment (tonistiigi) [inline CHANGELOG.md]: The problem is in file extraction so not directly related to builder. Could probably happen in pull/load/cp as well in certain conditions. I'd add "during extraction ...". Or "Properly handle shared mount propagation in storage directory".

Comment (vieux) [inline CHANGELOG.md]: I like `Properly handle shared mount propagation in storage directory`

Comment (thaJeztah) [inline CHANGELOG.md]: SGTM

**Comment by vieux** [comment]
please take a look @aaronlehmann @mrjana @riyazdf

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,70 @@ information on the list of deprecated flags and APIs please have a look at
...
+- Prevent a deadlock in libcontainerd for Windows [#27136](https://github.com/docker/docker/pull/27136)
+- Fix error reporting in CopyFileWithTar [#27075](https://github.com/docker/docker/pull/27075)
+* Reset health status to starting when a container is restarted [#27387](https://github.com/docker/docker/pull/27387)
+* Properly handle shared mount propagation in storage directory [#27609](https://github.com/docker/docker/pull/27609)
+- Fix docker exec [#27610](https://github.com/docker/docker/pull/27610)
+
+### Swarm Mode
+
+- Fix conversion of restart-policy [#27062](https://github.com/docker/docker/pull/27062)
+* Vendor Swarmkit [#27554](https://github.com/docker/docker/pull/27554)
```
Comment (icecrime) [inline CHANGELOG.md]: Maybe "Update" instead of "Vendor"?

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,70 @@ information on the list of deprecated flags and APIs please have a look at
...
+ * Avoid restarting a task that has already been restarted [https://github.com/docker/swarmkit/pull/1305]
```
Comment (icecrime) [inline CHANGELOG.md]: I don't know if that's intentional, but this isn't a properly formatted markdown link.

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,70 @@ information on the list of deprecated flags and APIs please have a look at
...
+ * Allow multiple randomly assigned published ports on service [https://github.com/docker/swarmkit/pull/1657]
+ - Fix panic when allocations happen at init time [https://github.com/docker/swarmkit/pull/1651]
+
+### Networking
+
+* Vendor libnetwork [#27559](https://github.com/docker/docker/pull/27559)
+ - Fix race in serializing sandbox to string [docker/libnetwork#1495](https://github.com/docker/libnetwork/pull/1495)
+ - Fix race during deletion [docker/libnetwork#1503](https://github.com/docker/libnetwork/pull/1503)
+ * Reset endpoint port info on connectivity revoke in bridge driver [docker/libnetwork#1504](https://github.com/docker/libnetwork/pull/1504)
+
```
Comment (mrjana) [inline CHANGELOG.md]: Add the following as well:
   * Fix a deadlock in networking code [docker/libnetwork#1507](https://github.com/docker/libnetwork/pull/1507)
   * Fix a race in load balancer state [docker/libnetwork#1512](https://github.com/docker/libnetwork/pull/1512)

**Review by icecrime — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,70 @@ information on the list of deprecated flags and APIs please have a look at
...
+- Remove the `Requires=docker.socket` directive from the `/usr/lib/systemd/system/docker.service` file if present
+- Remove `-H fd://` from the `ExecStart` directive (both in the main unit file, and in any drop-in files present).
+
+After making those changes, run `sudo systemctl daemon-reload`, and `sudo
+systemctl restart docker` to reload changes and (re)start the docker daemon.
+
+
+### Runtime
+
+- Fix user permissions in containers (CVE-2016-8867) [#27610](https://github.com/docker/docker/pull/27610)
```
Comment (riyazdf) [inline CHANGELOG.md]: s/Fix user permissions in containers/Fix ambient capability usage in containers/

**Review by riyazdf — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 6 above)
Comment (AkihiroSuda) [inline CHANGELOG.md]: CVE-2016-8867 not published yet?


http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-8867

File: CHANGELOG.md (hunk 4 above)
Comment (vieux) [inline CHANGELOG.md]: thanks

**Comment by vieux** [comment]
@icecrime @riyazdf PTAL

**Comment by riyazdf** [comment]
LGTM

**Comment by icecrime** [comment]
LGTM

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,71 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+### Networking
+
+* Update libnetwork [#27559](https://github.com/docker/docker/pull/27559)
+ - Fix race in serializing sandbox to string [docker/libnetwork#1495](https://github.com/docker/libnetwork/pull/1495)
+ - Fix race during deletion [docker/libnetwork#1503](https://github.com/docker/libnetwork/pull/1503)
+ * Reset endpoint port info on connectivity revoke in bridge driver [docker/libnetwork#1504](https://github.com/docker/libnetwork/pull/1504)
+ - Fix a deadlock in networking code [docker/libnetwork#1507](https://github.com/docker/libnetwork/pull/1507)
+ - Fix a race in load balancer state [docker/libnetwork#1512](https://github.com/docker/libnetwork/pull/1512)
+### Logging
```
Comment (icecrime) [inline CHANGELOG.md]: Minor nit: missing new line.

**Review by icecrime — APPROVED** [review]

File: CHANGELOG.md (hunk 7 above)
Comment (vieux) [inline CHANGELOG.md]: thanks.

**Review by riyazdf — APPROVED** [review]
LGTM

**Pull request merged**